docs/source/en/generation_strategies.mdx

```diff
@@ -0,0 +1,287 @@
+>>> the most efficient and sustainable human habitation and food production systems. \
+>>> Permaculture is a design system that encompasses a wide variety of disciplines, such \
+>>> as ecology, landscape design, environmental science and energy conservation, and the \
+>>> Permaculture design principles are drawn from these various disciplines. Each individual \
+>>> design principle itself embodies a complete conceptual framework based on sound \
+>>> scientific principles. When we bring all these separate  principles together, we can \
+>>> create a design system that both looks at whole systems, the parts that these systems \
+>>> consist of, and how those parts interact with each other to create a complex, dynamic, \
+>>> living system. Each design principle serves as a tool that allows us to integrate all \
+>>> the separate parts of a design, referred to as elements, into a functional, synergistic, \
+>>> whole system, where the elements harmoniously interact and work together in the most \
+>>> efficient way possible."
+
+>>> tokenizer = AutoTokenizer.from_pretrained(checkpoint)
+>>> input_ids = tokenizer(prompt, return_tensors="pt").input_ids
+
+>>> model = AutoModelForSeq2SeqLM.from_pretrained(checkpoint)
+
+>>> outputs = model.generate(input_ids, num_beams=5, num_beam_groups=5, max_new_tokens=30)
+>>> tokenizer.decode(outputs[0], skip_special_tokens=True)
+'The Design Principles are a set of universal design principles that can be applied to any location, climate and culture, and they allow us to design the most efficient and sustainable human habitation and food production systems.'
+```
+
+This guide illustrates the main parameters that enable various decoding strategies. More advanced parameters exist for the
+generate() method, which give you even further control over the generate() method's behavior.
+For the complete list of the available parameters, refer to the [API documentation](./main_classes/text_generation.mdx).
+
+## Save a custom decoding strategy with your model
+If you would like to share your fine-tuned model with a specific generation configuration, you can do so:
+* Create a `GenerationConfig` class instance
+* Specify the decoding strategy parameters
+* Save your generation configuration with `save_pretrained`, making sure to leave its `config_file_name` argument empty
+* Set `push_to_hub` to `True` to upload your config to the model's repo
+
+```python
+>>> from transformers import AutoModelForCausalLM, GenerationConfig
+
+>>> model = AutoModelForCausalLM.from_pretrained("my_account/my_model")
+>>> generation_config = GenerationConfig(
+...     max_new_tokens=50, do_sample=True, top_k=50, eos_token_id=model.config.eos_token_id
+... )
+>>> generation_config.save_pretrained("my_account/my_model", push_to_hub=True)
+```
+
+You can also store several generation parametrizations in a single directory, making use of the `config_file_name`
```

Comment:
```suggestion
You can also store several generation parameters in a single directory, making use of the `config_file_name`
```

Reply:
I think it should've been "several generation configurations".